node/lib/util/git_util.js

```diff
@@ -581,3 +581,130 @@ exports.parseRefspec = function(str) {
 
     throw new UserError("Refspec must match the format <src>:<dest>.");
 };
+
+/**
+ * Return the editor command to use for the specified `repo`.
+ *
+ * @param {NodeGit.Repository} repo
+ * @return {String}
+ */
+exports.getEditorCommand = co.wrap(function *(repo) {
+    assert.instanceOf(repo, NodeGit.Repository);
+    // TODO: libgit2 doesn't implement the equivalent of `git var` (or if it
+    // does I can't see where), so rather than code this myself I shell out to
+    // `git`.
+
+    const result =
+             yield ChildProcess.exec(`git -C '${repo.path()}' var GIT_EDITOR`);
+    return result.stdout.split("\n")[0];
+});
+
+/**
+ * Return the raw result of invoking the configured editor for the specified
+ * `repo` with a file containing the specified `initialContents`.  Note that
+ * the result may include `initialContents`; this method does not process the
+ * result in any way.
+ *
+ * @async
+ * @param {NodeGit.Repository} repo
+ * @param {String}             initialContents
+ * @return {String}
+ */
+exports.editMessage = co.wrap(function *(repo, initialContents) {
+    const messagePath = path.join(repo.path(), "COMMIT_EDITMSG");
+    yield fs.writeFile(messagePath, initialContents);
+    const editorCommand = yield exports.getEditorCommand(repo);
+
+    // TODO: if we ever need this to work on Windows, we'll need to do
+    // something else.  The `ChildProcess.exec` method doesn't provide for a
+    // way to auto-redirect stdio or I'd use it.
+
+    yield ChildProcess.spawn("/bin/sh",
```

Comment:
In git, if the editor exits with an error status, the commit does not happen.  I think this should happen here.